### PR TITLE
[Validator] Add `CompoundConstraintTestCase` to ease testing Compound constraints

### LINKED ISSUE #20148: [Validator] Add `CompoundConstraintTestCase` to ease testing Compound C…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#49547
| PR author(s) | @alexandre-daubois
| Merged in    | 7.2

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.2 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix #20148

May be tweaked a bit if https://github.com/symfony/symfony/pull/58064 passes.

### CODE REVIEW

**Comment by xabbuh** [comment]
@alexandre-daubois symfony/symfony#58064 has been merged

**Comment by alexandre-daubois** [comment]
Thanks! Updated

File: validation/custom_constraint.rst [hunk 1]
```diff
@@ -545,3 +548,74 @@ class to simplify writing unit tests for your custom constraints::
...
+Let's say you create a compound constraint that checks if a string meets
+your minimum requirements for your password policy::
+
+    // src/Validator/MatchPasswordPolicy.php
+    namespace App\Validator;
+
+    use Symfony\Component\Validator\Constraints as Assert;
+
+    #[\Attribute]
+    class MatchPasswordPolicy extends Assert\Compound
```
Comment (OskarStark) [inline validation/custom_constraint.rst]:
```suggestion
    class PasswordRequirements extends Assert\Compound
```
?

Comment (alexandre-daubois) [reply validation/custom_constraint.rst]: Here you go

**Comment by javiereguiluz** [comment]
Alex, another awesome contribution! You're on a roll! Thanks 🙇

**Pull request merged**